xbmc/platform/linux/SysfsPath.h

```diff
@@ -0,0 +1,72 @@
+/*
+ *  Copyright (C) 2011-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "utils/log.h"
+
+#include <fstream>
+#include <string>
+
+class CSysfsPath
+{
+public:
+  CSysfsPath() = default;
+  CSysfsPath(const std::string& path) : m_path(path) {}
+  ~CSysfsPath() = default;
+
+  void operator=(const std::string& path) { m_path = path; }
```

Comment:
I still think this is blatant abuse of `operator=` :-) (with a class named `CSysfsPath` it is unclear whether `=` should set the actual value or the path so I think it is best to avoid overriding this operator altogether)
My suggestion would be: Drop this function, create a fresh instance per path path.

Don't hold off merging because of this though, I'm just giving my opinion and it's nothing major.